Correction in configuring props doc (#418)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/configuring-properties.adoc

<configuration-properties file="${env}-properties.yaml"/>
----

> This way, the value of the property `env` determines which file to use to load the configuration properties. That `env` property can be set by a global property, system property, or environment property. Global properties are a way to define default values for properties, that can be overriden by any other type of property (e.g. a System Propery).

Reviewer: Suggested rewrite for the last sentence.

You can use global properties as a way to define default values for configuration properties. System and environment properties with the same name as a global property will override that global property.

Author: ```suggestion
This way, the value of the property `env` determines which file to use to load the configuration properties. That `env` property can be set by a global property, system property, or environment property. You can use global properties as a way to define default values for configuration properties. System and environment properties with the same name as a global property will override that Global Property.
```

Reviewer: Please make Global Property lowercase (the last words in the sentence.)

> This way, the value of the property `env` determines which file to use to load the configuration properties. That `env` property can be set by a global property, system property, or environment property. You can use global properties as a way to define default values for configuration properties. System and environment properties with the same name as a global property will override that Global Property.

Reviewer: Please make Global Property lowercase in the last sentence.

Author: ```suggestion
This way, the value of the property `env` determines which file to use to load the configuration properties. That `env` property can be set by a global property, system property, or environment property. You can use global properties as a way to define default values for configuration properties. System and environment properties with the same name as a global property will override that global property.
```

. System Properties
. Environment Properties
. Deployment Properties

> . Application Properties (includes Configuration Properties, Secure Configuration Properties and other Custom Configuration Properties)

Reviewer: I suggest this rewrite for the Hierarchies section:

Configuration properties can be overwritten. The system evaluates these properties in the following order:
. System properties
. Environment properties
. Deployment properties
. Application properties
. Global properties

So, for example, if a configuration property named 'size' is defined as both a system property and an application property, the value of the application property named 'size' will override that of the system property named 'size'.

Also, the value of an application property can depend on system, environment, and/or deployment properties. The value of deployment properties can depend on environment and/or system properties, and so on. For example, if there is a system property named 'env', a configuration property can have the value 'file.${env}.xml'.

> . Global Properties

Reviewer: Please change the Hierarchies section to this text.

Configuration properties can be overwritten. The system evaluates these properties in the following order:
. System properties
. Environment properties
. Deployment properties
. Application properties
. Global properties

So, for example, if a configuration property named 'size' is defined as both a system property and an application property, the value of the application property named 'size' will override that of the system property named 'size'.

Also, the value of an application property can depend on the values of properties that are higher up int the hierarchy. The value of application properties  can derive from deployment, enviornment, and/or system properties. The value of deployment properties can derive from environment and/or system properties, and so on. For example, if there is a system property named 'env', a configuration property can have the value 'file.${env}.xml'.

Author: > the value of the application property named 'size' will override that of the system property named 'size'

This is the opposite of what actually happens, as stated here:
> the value for that application is the one with the most precedence (in this case, the System Property).

Reviewer: Properties Hierarchy
Configuration properties can be overwritten. The Mule runtime engine uses the following hierarchy to determine which properties take precedence if they have the same name. In this hierarchy, system properties are at the top, so they take precedence over the other types of properties.

. System properties

. Environment properties

. Deployment properties

. Application properties (includes configuration properties, secure configuration properties and other custom configuration properties)

. Global Properties

So, for example, if a configuration property named 'size" is defined as a system property, and there is also an application configuration property named 'size', the value for the application is the value of the property with the most precedence (in this case, the system property).

Also, a property can derive its value from properties that are higher in the hierarchy. Therefore, an application property can derive its value from environment, system, and/or deployment properties. A deployment property can derive its value from an environment and/or system property, and so on. For example, if there is a system property named 'env', an application configuration property can have the value 'file.$github.xml'.

> 

Reviewer: So, for example if a configuration property is defined as a system property, and there is also an application configuration of the same name, the application uses the value of the system property, since that is evaluated before the application property.

So, for example, if a configuration property is defined in a System Property, and there is also an application configuration property, the value for that application is the one with the most precedence (in this case, the System Property).

> Also, an application property could depend its value on Environment, System and/or Deployment properties. Deployment Properties could depend on System Properties, and so on. For example, if there is a System Property named `env`, a Configuration Property could have its value `file.${env}.xml`. 

Reviewer: Also, the value of an application property can depend on environment, system, and/or deployment properties. The value of deployment properties can depend on environment and/or system properties and so on. For example, if there is a system property named `env`, a configuration property can have the value `file.${env}.xml`

. Environment Properties
. Deployment Properties

> . Application Properties

Author: ```suggestion
. Application Properties (includes Configuration Properties, Secure Configuration Properties and other Custom Configuration Properties)
```

. Global Properties

> So, for example, if a configuration property is defined in a System Property, and there is also an application configuration property, the value for that application is the first one.

Reviewer 2: Not sure this is clear for the end user. Is the concept of **application configuration property** clear?. Also instead of talking about the **first one** I would talk about precedence.

Author: ```suggestion
So, for example, if a configuration property is defined in a System Property, and there is also an application configuration property, the value for that application is the most precedence (in this case, the System Property).
```
